src/keystore/persistence/EncryptedPersistence.ts

```diff
@@ -0,0 +1,104 @@
+import { Persistence } from './interface'
+import eccrypto, { Ecies } from 'eccrypto'
+
+const IV_LENGTH = 16
+const EPHEMERAL_PUBLIC_KEY_LENGTH = 65
+const MAC_LENGTH = 32
+const AES_BLOCK_SIZE = 16
+const MIN_DATA_LENGTH =
+  IV_LENGTH + EPHEMERAL_PUBLIC_KEY_LENGTH + MAC_LENGTH + AES_BLOCK_SIZE
+
+const assertEciesLengths = (ecies: Ecies): void => {
+  if (ecies.iv.length !== IV_LENGTH) {
+    throw new Error('Invalid iv length')
+  }
+  if (ecies.ephemPublicKey.length !== EPHEMERAL_PUBLIC_KEY_LENGTH) {
+    throw new Error('Invalid ephemPublicKey length')
+  }
+  if (
+    ecies.ciphertext.length < 1 ||
+    ecies.ciphertext.length % AES_BLOCK_SIZE !== 0
+  ) {
+    throw new Error('Invalid ciphertext length')
+  }
+  if (ecies.mac.length !== MAC_LENGTH) {
+    throw new Error('Invalid mac length')
+  }
+}
+
+const serializeEcies = (ecies: Ecies): Uint8Array => {
+  assertEciesLengths(ecies)
+  const { iv, ephemPublicKey, ciphertext, mac } = ecies
+  const result = new Uint8Array(
+    iv.length + ephemPublicKey.length + ciphertext.length + mac.length
+  )
+  result.set(iv, 0)
+  result.set(ephemPublicKey, IV_LENGTH)
+  result.set(ciphertext, IV_LENGTH + EPHEMERAL_PUBLIC_KEY_LENGTH)
+  result.set(mac, IV_LENGTH + EPHEMERAL_PUBLIC_KEY_LENGTH + ciphertext.length)
```

Comment:
I wasn't able to find a nice existing library. We _could_ do the serialization by creating a Protobuf message for ECIES data, but given all the fields but one are of a fixed length it feels unnecessary. We'd still need to validate the lengths of all the fields and otherwise validate the data.